### PR TITLE
Users are now able to search through messages on their profile pages

### PR DESCRIPTION
**Pull request by sarakvaska**

### CODE REVIEW

File: src/main/webapp/WEB-INF/view/profile.jsp [hunk 1]
```diff
@@ -86,22 +110,39 @@ List<Message> messages = (List<Message>) request.getAttribute("messages");
           </form>
           <button type ="submit" name="about" value="About">Submit</button>
           <h3><%= getProfile.getName() %>'s Sent Messages</h3>
-          <div id = "chat">
-            <ul>
+          <input type="text" id="searchInput" onkeyup="mySearch()" placeholder="Search for messages...">
```
Comment (aljonpineda) [inline src/main/webapp/WEB-INF/view/profile.jsp]: What does `onkeyup` do?

File: src/main/webapp/WEB-INF/view/profile.jsp [hunk 2]
```diff
@@ -21,8 +32,68 @@
...
+        function mySearch() {
+            var input, filter, listMessage, li, a, i;
+            input = document.getElementById("searchInput");
+            filter = input.value.toUpperCase();
+            listMessage = document.getElementById("messageList");
+            indivMessage = listMessage.getElementsByTagName("li");
+            for (i = 0; i < indivMessage.length; i++) {
+                a = indivMessage[i].getElementsByTagName("a")[0];
+                if (a.innerHTML.toUpperCase().indexOf(filter) > -1) {
+                    indivMessage[i].style.display = "";
```
Comment (aljonpineda) [inline src/main/webapp/WEB-INF/view/profile.jsp]: Does setting the display to `""` keep the individual message visible?

Comment (sarakvaska) [reply src/main/webapp/WEB-INF/view/profile.jsp]: Yes

File: src/main/webapp/WEB-INF/view/profile.jsp (hunk 1 above)
Comment (sarakvaska) [reply src/main/webapp/WEB-INF/view/profile.jsp]: onkeyup is like onclick except that it works as the user types in their input into the search bar, whereas with onclick the user presses a button and then the function is called

**Comment by cindylee0113** [comment]
Just wondering, is the pull request fine on its own or does it need other pull requests to be merged in first prior to running the server? Because I tried testing, and I'm seeing errors

**Pull request closed without merging**